Dockerfile

```diff
@@ -1,69 +1,96 @@
+# syntax=docker.io/docker/dockerfile:1.7-labs
 # Pelican Production Dockerfile
 
 # ================================
-# Stage 1: Build PHP Dependencies
+# Stage 1: Build PHP Base Image
 # ================================
-FROM --platform=$TARGETOS/$TARGETARCH php:8.3-fpm-alpine AS composer
+FROM --platform=$TARGETOS/$TARGETARCH php:8.3-fpm-alpine AS base
+
+ADD --chmod=0755 https://github.com/mlocati/docker-php-extension-installer/releases/latest/download/install-php-extensions /usr/local/bin/
+
+RUN install-php-extensions bcmath gd intl zip opcache pcntl posix pdo_mysql
+
+RUN rm /usr/local/bin/install-php-extensions
+
+# ================================
+# Stage 2-1: Composer Install
+# ================================
+FROM --platform=$TARGETOS/$TARGETARCH base AS composer
 
 WORKDIR /build
 
-COPY . ./
 COPY --from=composer:latest /usr/bin/composer /usr/local/bin/composer
 
-# Install required libraries and PHP extensions
-RUN apk update && apk add --no-cache \
-    libpng-dev libjpeg-turbo-dev freetype-dev libzip-dev icu-dev \
-    zip unzip curl \
-    && docker-php-ext-install bcmath gd intl zip opcache pcntl posix pdo_mysql
+# Copy bare minimum to install Composer dependencies
+COPY composer.json composer.lock ./
 
-RUN composer install --no-dev --optimize-autoloader
+RUN composer install --no-dev --no-interaction --no-autoloader --no-scripts
```

Comment:
Optimize should stay

Reply:
This is just the install phase, where composer installs dependencies. Optimize is run at the optimize phase below.